### PR TITLE
SAML configuration changes to handle IDP initiated SSO

### PR DESCRIPTION
**Pull request by amirinenik**
**SAML configuration changes to handle IDP initiated SSO**
1. Process IDP initiated login SAML response by ignoring validation of the "InResponseTo" property value

```
    <!-- Provider of default SAML Context to process IDP initiated login SAML response-->
    <b:bean id="contextProvider" class="org.springframework.security.saml.context.SAMLContextProviderImpl">
        <b:property name="storageFactory">
            <b:bean class="org.springframework.security.saml.storage.EmptyStorageFactory"/>
        </b:property>
    </b:bean>
```

2. Use relayState URL from SAML response to navigate to that URL upon successful authentication

```
    <!--
        RelayState success handler to process return URL for IDP initiated login SAML response.
        Application landing page is going to be RelayState URL upon successfull authentication.
    -->
    <b:bean id="successRedirectHandler" class="org.springframework.security.saml.SAMLRelayStateSuccessHandler">
        <b:property name="defaultTargetUrl" value="/" />
    </b:bean>
```

### CODE REVIEW

**Comment by amirineni1** [comment]
@pvannierop I agree that this feature should be configurable for security reasons. The default SAML authentication behavior should be SP initiated. I made a few changes to support configurable SAML context that would allow configuring session storage or no storage to support SP/IdP initiated logins.  

Added new property to the portal.properties file. Default authentication behavior for SAML is SP initiated because session storage is enabled to check for request id when SAML response is received.
```
# Default SAML Context
# Use org.springframework.security.saml.storage.EmptyStorageFactory class as default context for IDP initiated logins
saml.context.default=org.springframework.security.saml.storage.HttpSessionStorageFactory
```

Property is consumed in the application security file to enable the session or no storage for SP/IdP logins.

```
    <!-- Provider of default SAML Context -->
    <b:bean id="contextProvider" class="org.springframework.security.saml.context.SAMLContextProviderImpl">
        <b:property name="storageFactory">
            <b:bean class="${saml.context.default}"/>
        </b:property>
    </b:bean>
```

**Comment by amirineni1** [comment]
@pvannierop modified configuration to support backward compatibility changes.

**Comment by amirineni1** [comment]
@pvannierop I agree, default context already set in the security configuration file so saml.context property for change the context. Renamed the property.

File: security/security-spring/src/main/resources/applicationContext-security.xml [hunk 1]
```diff
@@ -401,7 +401,19 @@
...
+            <b:bean class="${saml.context:org.springframework.security.saml.storage.HttpSessionStorageFactory}"/>
+        </b:property>
+    </b:bean>
+
+    <!--
+        RelayState success handler to process return URL for IDP initiated login SAML response.
+        Application landing page is going to be RelayState URL upon successfull authentication.
+    -->
+    <b:bean id="successRedirectHandler" class="org.springframework.security.saml.SAMLRelayStateSuccessHandler">
+        <b:property name="defaultTargetUrl" value="/" />
```
Comment (n1zea144) [inline security/security-spring/src/main/resources/applicationContext-security.xml:415]: Should this defaultTargetUrl be consistent with [successRedirectHandler](https://github.com/cBioPortal/cbioportal/blob/6a5fbbbb141a4850fd36f10222ff81592c5e5484/security/security-spring/src/main/resources/applicationContext-security.xml#L125).defaultTargetUrl ?

Comment (amirineni1) [reply security/security-spring/src/main/resources/applicationContext-security.xml:415]: Yes, it is. Application behavior remains the same for defaultTargetURL. The application always redirects to the user requested URL upon successful login. SAMLRelayStateSuccessHandler additionally process relayState property URL part of SAML response from the IDP.

**Pull request closed without merging**